flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/service/materializedtable/MaterializedTableManager.java

```diff
@@ -169,6 +181,83 @@ private static void createMaterializedInContinuousMode(
         }
     }
 
+    private static ResultFetcher callAlterMaterializedTableRefreshOperation(
+            OperationExecutor operationExecutor,
+            OperationHandle handle,
+            AlterMaterializedTableRefreshOperation alterMaterializedTableRefreshOperation) {
+        ObjectIdentifier materializedTableIdentifier =
+                alterMaterializedTableRefreshOperation.getTableIdentifier();
+        ResolvedCatalogBaseTable<?> table = operationExecutor.getTable(materializedTableIdentifier);
+        if (!(table instanceof ResolvedCatalogMaterializedTable)) {
+            throw new TableException(
+                    String.format(
+                            "The table '%s' is not a materialized table.",
+                            materializedTableIdentifier));
+        }
+
+        ResolvedCatalogMaterializedTable materializedTable =
+                (ResolvedCatalogMaterializedTable) table;
+
+        Map<String, String> partitionSpec =
+                alterMaterializedTableRefreshOperation.getPartitionSpec();
+
+        Set<String> allPartitionKeys =
+                new HashSet<>(((ResolvedCatalogMaterializedTable) table).getPartitionKeys());
+        Set<String> unknownPartitionKeys = new HashSet<>(partitionSpec.keySet());
+        unknownPartitionKeys.removeAll(allPartitionKeys);
+        if (!unknownPartitionKeys.isEmpty()) {
```

Comment:
Only this check is not enough. Here I need to add some more context. When creating a Materialized Table, the specified partition key field may contain both String and non-String types. Suppose `p1` is string and `p2` is int. The user writes the following refresh statement:
```
ALTER MATERIALIZED TABLE t1 REFRESH PARTITION(p1 = '20240507', p2 = 2)
```
Although p2 is not a string type, we uniformly converted it to string type to handle it at the time of Parser.

Based on the above background, there are two strategies for handling WHERE conditions when splicing INSERT statements:
1. the first strategy is to check the type of the partition key field, if it is not string type, directly thrown an error, that is, does not support the Refresh partition specified in the non-String type of the partition key. in fact, this demand should also be rare
2. when splicing WHERE conditions, for String type fields, you need to splice into “%s = ‘%s’”, non-String type splice into “%s = %s”, of course, I'm not sure if this can completely cover all non-String types.

Reply:
Choose 1. We can only support string type at first.